### ISSUE TITLE
Bulk Image Processing using find

### ISSUE THREAD

**Issue by rahijain** [Lymphatus/caesium-clt#17]
I am trying to install it on servers to process 100Gb+ of images on recurring basis. With Jpegoptim i have used - 

find /web/html/skin/ -name '*.jpg' | xargs jpegoptim --strip-all --all-progressive

When i am using the same with 

find /web/html/skin/ -name '*.jpg' | xargs -0 caesiumclt -q 90 -o /web/html/skin/ 

Its not able to take the parameters. Error coming - 
xargs: unmatched single quote; by default quotes are special to xargs unless you use the -0 option
[ERROR] 6: Cannot check if is a directory.

Any way to process in bulk using find? Let me know what i am doing wrong here?

**Comment by Lymphatus** [comment]
The program does not support this kind of syntax yet.
I'm going to add it in the next release.

A side note: for security reasons, since the program is still beta, it does not allow to compress images in-place. If you set the output folder equal to the input one, it throws an error. It will be removed when is more stable.

**Comment by VanKusanagi** [comment]
So, the same issue occured for me as well. But I realised that the destination and source directories should be absolute(/home/user/destination or ~/destination) or it won't  work. It somehow seems that it does not have an idea the directory thecommand is being executed in
![image](https://user-images.githubusercontent.com/13829905/32364451-f6b8e9ea-c09a-11e7-8d5e-6dfdb9269c44.png)

**Comment by Lymphatus** [comment]
Yes, it seems it does not recognize the folder you are in, but works fine for absolute paths. I'm going to address this issue right away.

**Comment by VanKusanagi** [comment]
While you are at it, can you also change it to support whitespaces and escape sequences in file names?

**Comment by VanKusanagi** [comment]
Sorry, my bad. I think it does support whitespaces

**Comment by Lymphatus** [comment]
Sure. In my testing branch I solved the issue with relative paths not being resolved, but I'm facing some other small problems I will get rid off soon. I will also test escape sequnces, and the Windows platform, which manages paths in a quite different way.

**Comment by VanKusanagi** [comment]
I think it is relevant to this thread so I am starting this. I tried using this tool for optimizing images on an S3 bucket and post optimization transfer it to another bucket. 
The source bucket contains atleast 10 million files while the destination is empty
So, after running the process using -RS it would take 6-8 hrs even before starting which I believe is because it prepares an array/list of images to optimize beforehand. Even then after the process started, it would sometimes throw an error about "Not being able to calculate size of a file" and terminate. In my recommendation it should just place the file path it encountered the error and love on

**Comment by Lymphatus** [comment]
The latter seems a different problem than the one addressed here. I suggest you to create a new issue facing this problem, so we can discuss it there.

**Comment by VanKusanagi** [comment]
The first letter of path/directory issue didn't occur for me when I was using -RS. The issue I am stating is:
"Not able to calculate size of file" at some point during optimization.

**Comment by Lymphatus** [comment]
> I am trying to install it on servers to process 100Gb+ of images on recurring basis. With Jpegoptim i have used -
> 
> find /web/html/skin/ -name '*.jpg' | xargs jpegoptim --strip-all --all-progressive
> 
> When i am using the same with
> 
> find /web/html/skin/ -name '*.jpg' | xargs -0 caesiumclt -q 90 -o /web/html/skin/
> 
> Its not able to take the parameters. Error coming -
> xargs: unmatched single quote; by default quotes are special to xargs unless you use the -0 option
> [ERROR] 6: Cannot check if is a directory.
> 
> Any way to process in bulk using find? Let me know what i am doing wrong here?

I went through some testing and I could use the find syntax by not using the -0 argument to xargs. Also, by passing files one by one, the program cannot keep the folder structure as it's a feature enabled by passing a folder as input.
